src/neptune/common/backends/utils.py

```diff
@@ -120,10 +132,21 @@ def wrapper(*args, **kwargs):
                     HTTPBadGateway.status_code,
                     HTTPServiceUnavailable.status_code,
                     HTTPGatewayTimeout.status_code,
-                    HTTPTooManyRequests.status_code,
                     HTTPInternalServerError.status_code,
                 ):
-                    time.sleep(min(2 ** min(10, retry), MAX_RETRY_TIME))
+                    time.sleep(min(2 ** min(MAX_RETRY_MULTIPLIER, retry), MAX_RETRY_TIME))
+                    last_exception = e
+                    continue
+                elif status_code == HTTPTooManyRequests.status_code:
+                    try:
+                        wait_time = (
+                            int(e.response.headers["retry-after"][0])
+                            if "retry-after" in e.response.headers
+                            else 2 ** min(MAX_RETRY_MULTIPLIER, retry)
+                        )
+                        time.sleep(min(wait_time, MAX_RETRY_TIME))
```

Comment:
Can we use `time.sleep` only once?

Reply:
Probably here we should only have something like `time.sleep(min(get_retry_from_headers_or_default(e.response), MAX_RETRY_TIME))`

Reply:
This is a great idea, I added this to the code now! :D

Reply:
This is a great idea, I added this to the code now! :D

Reply:
This is a great idea, I added this to the code now! :D